chain/client/src/client.rs

```diff
@@ -2448,6 +2542,23 @@ impl Client {
         Ok(false)
     }
 
+    fn set_state_sync_hash<'a>(
+        mut update: ChainStoreUpdate<'a>,
+        epoch_first_block: CryptoHash,
+        state_sync_info: &mut StateSyncInfo,
```

Comment:
Why does this need to be mutable?

Reply:
i did this just as a helper to set the in-memory and on-disk values at the same time, but actually let's just remove it. I think I thought I would need it in more than one place when I put it there